indsl/fluid_dynamics/friction.py

```diff
@@ -16,11 +19,375 @@ def Haaland(Re: pd.Series, roughness: float) -> pd.Series:
     the accuracy of the data.
 
     Args:
-        Re: Reynolds Number
-        roughness: Surface roughness
+        Re: Reynolds Number [-].
+        roughness: Surface roughness [m].
 
     Returns:
-        pandas.Series: Friction factor
+        pandas.Series: Friction factor [-]
     """
     den = -1.8 * np.log10((6.9 / Re[Re != 0]) + (roughness / 3.7) ** 1.1)
     return 1 / den**2
+
+
+@check_types
+def Colebrook(
+    Re: Union[pd.Series, float],
+    roughness_scaled: Union[pd.Series, float],
+    align_timesteps: bool = False,
+) -> pd.Series:
+    r"""Colebrook approximation.
+
+    A computationally efficient Colebrook approximation taken from the following article
+
+     "Fast and Accurate Approximations for the Colebrook Equation"
+     Dag Biberg
+     Journal of Fluids Engineering Copyright VC 2017 by ASME MARCH 2017, Vol. 139 / 031401-1
+
+    This is a faster and more accurat approximation to Colebrook than Haaland
+
+    Args:
+        Re: Reynolds Number [-].
+        roughness_scaled: Scaled surface roughness [-].
+            The wall surface roughness is normally scaled with the pipe inner diameter
+        align_timesteps: Auto-align.
+            Automatically align time stamp  of input time series. Default is False.
+
+    Returns:
+        pandas.Series: Friction factor [-]
+    """
+    from math import log as m_log
+
+    # Every Series input needs to be autaligned, unless opted out
+    if align_timesteps:
+        Re, roughness_scaled = auto_align([Re, roughness_scaled], align_timesteps)
+
+    a = 2 / m_log(10)
+    b = 2.51
+    c = 3.7
+    ab = a * b
+
+    logReab = np.log(Re / ab)
+    x = logReab + Re * roughness_scaled / (ab * c)
+    G = np.log(x) * (1 / x - 1)
+    denominator = a * (logReab + G)
+    friction_factor = (1 / denominator) ** 2
+
+    if not isinstance(friction_factor, pd.Series):
+        friction_factor = pd.Series([friction_factor])
+    return friction_factor
+
+
+@check_types
+def friction_factor_laminar(Re: Union[pd.Series, float]) -> pd.Series:
+    r"""Friction factor laminar flow.
+
+    friction_factor = :math:`\frac{64}{Re}`
+        Re: Reynolds number = :math:`\frac{\rho u D}{\mu}`
+
+    Args:
+        Re: Reynolds Number [-].
+
+    Returns:
+        pandas.Series: Friction factor laminar flow [-]
+    """
+    friction_factor = 64 / Re
+    if not isinstance(friction_factor, pd.Series):
+        friction_factor = pd.Series([friction_factor])
+    return friction_factor
+
+
+@check_types
```

Comment:
You dont have to add checktypes on the private functions :)

Reply:
Good point